requirements.txt

```diff
@@ -5,18 +5,18 @@ boto3>=1.4.4
 cffi>=1.11.5
 configparser>=3.5.0
 croniter>=0.3.16
-elasticsearch>=7.0.0
+elasticsearch==7.0.0
```

Comment:
I made the following changes in yelp/elastalert, so I fixed it.

https://github.com/Yelp/elastalert/pull/2593#issue-349637490
>Also note: I also added a pin for elasticsearch==7.0.0, because apparently 7.1.0 will NOT work with ES < 6.6 due to it not supported _source_include(s?). 7.0.0 does. Tests won't pass otherwise.


If `elasticsearch>=7.0.0,<8.0.0`, the warning of include_type_name will be issued.
I think that the new version of elasticsearch-py has more functions to warn of deprecated ones in the next major version. If the warning of include_type_name appears in 7.x in general, it is necessary to delete the description of include_type_name, but the correction part has not been specified. Also, since doc_type disappears in elasticsearch 8, I think that the warning of doc_type will be issued and the number of inquiries from users will increase. So I can understand the feeling very much, but at the moment I think that `elasticsearch==7.0.0` should be set.

```
16:13:49.571Z ERROR elastalert-server:
    ProcessController:  /home/node/.local/lib/python3.8/site-packages/elasticsearch/connection/base.py:190: ElasticsearchDeprecationWarning: Camel case format name dateOptionalTime is deprecated and will be removed in a future version. Use snake case name date_optional_time instead.
      warnings.warn(message, category=ElasticsearchDeprecationWarning)
    /home/node/.local/lib/python3.8/site-packages/elasticsearch/connection/base.py:190: ElasticsearchDeprecationWarning: [types removal] Using include_type_name in put mapping requests is deprecated. The parameter will be removed in the next major version.
      warnings.warn(message, category=ElasticsearchDeprecationWarning)
```

Reply:
Thanks for the detailed explanation. I'll get this merged in.